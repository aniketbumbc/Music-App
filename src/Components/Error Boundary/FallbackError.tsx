import React from 'react';
import './FallbackError.css';

const Fallback: React.FC = () => {
  return (
    <div>
      <h1> Something went wrong</h1>
      <h3>Please try again </h3>
    </div>
  );
};

export default Fallback;
