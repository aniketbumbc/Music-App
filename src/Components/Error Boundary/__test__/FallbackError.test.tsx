import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import FallbackError from '../FallbackError';

describe('FallbackError', () => {
  afterEach(cleanup);

  it('should have h1 and h3 element', () => {
    render(<FallbackError />);

    const h1Element = screen.getByRole('heading', { level: 1 });
    const h3Element = screen.getByRole('heading', { level: 3 });

    expect(h1Element).toBeInTheDocument();
    expect(h3Element).toBeInTheDocument();
  });

  it('Should  have correct text for h1 and h3 element', () => {
    render(<FallbackError />);

    const h1Element = screen.getByText('Something went wrong');
    const h3Element = screen.getByText('Please try again');

    expect(h1Element).toBeInTheDocument();
    expect(h3Element).toBeInTheDocument();
  });

  it('Shoud not have element in the DOM', () => {
    render(<FallbackError />);

    const nullElement = screen.queryByText('Error');

    expect(nullElement).not.toBeInTheDocument();
  });

  it('should take a snapshot of component', () => {
    const { asFragment } = render(<FallbackError />);

    expect(asFragment()).toMatchSnapshot();
  });
});
