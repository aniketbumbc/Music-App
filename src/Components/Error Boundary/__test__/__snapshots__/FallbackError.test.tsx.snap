// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FallbackError should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
       Something went wrong
    </h1>
    <h3>
      Please try again 
    </h3>
  </div>
</DocumentFragment>
`;

exports[`FallbackError should take a snapshot of component 1`] = `
<DocumentFragment>
  <div>
    <h1>
       Something went wrong
    </h1>
    <h3>
      Please try again 
    </h3>
  </div>
</DocumentFragment>
`;
